app/scripts/models/auth_brokers/pairing/authority.js

```diff
@@ -0,0 +1,110 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// Manages the OAuth flow by webchannel messages to the browser,
+// to help with a pairing-based flow.
+
+import AuthorityStateMachine from '../../pairing/auth-state-machine';
+import OAuthAuthenticationBroker from '../oauth';
+import setRemoteMetaData from './remote-metadata';
+
+export default class AuthorityBroker extends OAuthAuthenticationBroker {
+  type = 'pair-hybrid';
+
+  initialize (options) {
+    super.initialize(options);
+
+    const {  notifier } = options;
+
+    this.stateMachine = new AuthorityStateMachine({}, {
+      broker: this,
+      channelServerClient: this.channelServerClient,
+      notifier,
+      relier: this.relier,
+      useWebChannel: true
+    });
+
+
+    this.setCapability('requirePasswordToPair', false);
+  }
+
+  fetch () {
+    return Promise.resolve()
+      .then(() => super.fetch())
+      .then(() => this.getSupplicantMetadata());
+  }
+
+  start (/*account*/) {
+    // does nothing
+  }
+
+  _provisionScopedKeys() {
+    throw new Error('this should never be called');
+  }
+
+  getSupplicantMetadata() {
+    const remoteMetaData = this.get('remoteMetaData');
+    if (remoteMetaData) {
+      return Promise.resolve(remoteMetaData);
+    }
+
+    return this.request(this._notificationChannel.COMMANDS.PAIR_REQUEST_SUPPLICANT_METADATA)
+      .then((response) => {
+        console.log('got a response', response);
+        this.setRemoteMetaData(response);
+        return this.get('remoteMetaData');
+      });
+  }
+
+  setRemoteMetaData = setRemoteMetaData;
+
+  sendOAuthResultToRelier ({ code, state }) {
+    return this.send(channel.COMMANDS.PAIR_AUTHORIZE, { code, state });
+  }
+
+  getOAuthResult(account, options = {}) {
+    return Promise.resolve().then(() => {
+      const keysJwe = this.get('keysJwe');
+      if (keysJwe) {
+        return keysJwe;
+      }
+
+      return this.request(this._notificationChannel.COMMANDS.PAIR_REQUEST_KEYS_JWE);
+    }).then(keysJwe => {
+      this.set({ keysJwe });
```

Comment:
The desktop code was sending an object like `{ keys_jwe: JWE_VALUE }` but this code is expecting just `JWE_VALUE`; I've managed to get it working by changing desktop code to send just `JWE_VALUE` but TBH I think expecting an object feels cleaner...